browser/src/control/jsdialog/Widget.TreeView.ts

```diff
@@ -1064,6 +1072,41 @@ class TreeViewControl {
 		}, 100);
 	}
 
+	highlightEntries(searchTerm: string) {
+		app.highlightTerm = searchTerm;
+		if (this._filterTimer) clearTimeout(this._filterTimer);
+		var entriesToHighlight: Array<HTMLElement> = [];
+		var allEntries = this._container.querySelectorAll('.ui-treeview-entry');
+
+		searchTerm = searchTerm.trim();
+
+		allEntries.forEach((entry: HTMLElement) => {
+			if (searchTerm === '') return;
+
+			var cells = entry.querySelectorAll('div');
+			for (var i in cells) {
+				var entryText = cells[i].innerText;
+				if (
+					entryText &&
+					entryText.toLowerCase().indexOf(searchTerm.toLowerCase()) >= 0
+				) {
+					entriesToHighlight.push(entry);
+				}
+			}
+
+			return;
+		});
+
+		this._filterTimer = setTimeout(() => {
```

Comment:
Timeout is problematic (will happen in random moment when JS engine has time :) ) we rather prefer to not force browser to render additional frames for smooth experience.

To solve that problem we have queue: `app.layoutingService.appendLayoutingTask(() => { here your DOM touching code }` which is scheduled and processed when we acctually want to render

Reply:
also -> why we do only this class modification in the timer?

I would say it should be the whole current `highlightEntries` function content inside layouting task.
No additional wrapping of add/remove class